Purchases/Purchasing/Purchases.swift

```diff
@@ -1160,6 +1160,43 @@ public extension Purchases {
     }
 #endif
 
+    /**
+     * Use this function to retrieve a `Bool` if eligible for a given `StoreProductDiscount` for a given `StoreProduct`.
+     *
+     * - Parameter discount: The `StoreProductDiscount` to apply to the product.
+     * - Parameter product: The `StoreProduct` the user intends to purchase.
+     * - Parameter completion: A completion block that is called when the `Bool` is returned.
+     * If it was not successful, there will be an `Error`.
+     */
+    @available(iOS 12.2, macOS 10.14.4, macCatalyst 13.0, tvOS 12.2, watchOS 6.2, *)
+    @objc(checkEligiblilityForPromotionalOffer:product:completion:)
+    func checkEligiblilityForPromotionalOffer(forProductDiscount discount: StoreProductDiscount,
+                                              product: StoreProduct,
+                                              completion: @escaping (Bool, Error?) -> Void) {
+        if let sk1Product = product.sk1Product {
+            purchasesOrchestrator.promotionalOffer(forProductDiscount: discount,
```

Comment:
i'm not sure if this is possible or even worth it... but just playing devil's advocate... is it somehow possible to save the result here, so we don't have to make this call again if they try to purchase with it?

Reply:
Caching scares me a little bit 😬 It would inject a new place for issues to arise. I don't know how to decide on when to use the cache and when to invalidate it.

Do we just store the latest product and store product discount that was used? Or a bunch? Someone might call this method on a few different products so it seems like the cache would have to help multiple `PromotionalOffers` 🤔

Reply:
very fair. just thought we should think about it, let's consider it considered ✅

Reply:
🤔 is there any chance that when we try to create the offer the second time there's an issue?